include/ghex/structured/field_descriptor.hpp

```diff
@@ -0,0 +1,254 @@
+/* 
+ * GridTools
+ * 
+ * Copyright (c) 2014-2020, ETH Zurich
+ * All rights reserved.
+ * 
+ * Please, refer to the LICENSE file in the root directory.
+ * SPDX-License-Identifier: BSD-3-Clause
+ * 
+ */
+#ifndef INCLUDED_GHEX_STRUCTURED_FIELD_DESCRIPTOR_HPP
+#define INCLUDED_GHEX_STRUCTURED_FIELD_DESCRIPTOR_HPP
+
+#include <algorithm>
+#include "./field_utils.hpp"
+#include "../common/utils.hpp"
+#include "../arch_traits.hpp"
+#include <gridtools/common/array.hpp>
+#include <gridtools/common/layout_map.hpp>
+
+namespace gridtools {
+namespace ghex {
+namespace structured {
+    
+template<typename T, typename Arch, typename DomainDescriptor, int... Order>
+class field_descriptor
+{
+public: // member types
+    using value_type             = T;
+    using arch_type              = Arch;
+    using device_id_type         = typename arch_traits<arch_type>::device_id_type;
+    using domain_descriptor_type = DomainDescriptor;
+    using dimension              = std::integral_constant<std::size_t,sizeof...(Order)>;
+    using has_components         = std::integral_constant<bool,
```

Comment:
- in case the field is 3 dimensional (usual case), we have for example `field_descriptor<...,2,1,0>`, so 3 order arguments, this leads to dimension being `3`
    - the cubed sphere domain has a dimension of `3` (see `cubed_sphere::domain_descriptor::dimension`)
    - then `has_components = 3 > 3 = false`
- when we have a field with components we would use `field_descriptor<...,2,1,0,3>` for example
    - then `has_components = 4 > 3 = true`